### PR TITLE
ironbar: init at 0.13.0

### PR DESCRIPTION
**Pull request by donovanglover**
###### Description of changes

Adds https://github.com/JakeStanger/ironbar

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by teto** [comment]
quite happy to see a rust bar ! hpoing it would crash less than waybar

Seems like it is based on client/server module but the server seems to die pretty fast , i cant ping it.
```
2023-07-03T14:33:37.394681Z  INFO ironbar: 86: Ironbar version 0.13.0
2023-07-03T14:33:37.394701Z  INFO ironbar: 87: Starting application
2023-07-03T14:33:37.451658Z ERROR ironbar: 170: FileNotFound
2023-07-03T14:33:37.451659Z  INFO ironbar::ipc::server: 36: Starting IPC on /run/user/1000/ironbar-ipc.sock
```
I dont have a config so maybe that's the problem.

**Comment by donovanglover** [comment]
@teto Currently ironbar doesn't start without a default config. I ran into that issue as well when trying it for the first time. I made an issue here: https://github.com/JakeStanger/ironbar/issues/219

So far ironbar has suited my needs better than waybar. Rust makes it easier to contribute to and development is quite active as well.

**Comment by JakeStanger** [comment]
Hey, Ironbar author here, happy to see it get some love and excited to see it get packaged :)

I'll keep development discussion over in the repo, but just wanted to clarify:

> Seems like it is based on client/server module but the server seems to die pretty fast , i cant ping it.

There is a small IPC server that allows for clients to perform some external interactions, but it's otherwise unrequited. It's behind a feature flag so can be turned off if desired.

> Currently ironbar doesn't start without a default config

This is correct. It should be sufficient to stick an empty object `{}`  in `~/`config/ironbar/config.json` for now.

Give me a shout if you need anything, happy to help out here.

**Comment by teto** [comment]
> This is correct. It should be sufficient to stick an empty object {} in ~/config/ironbar/config.json for now

that indeed fixed it but a fancier default would be more enticing ;)

is the pkgs/applications/misc/ironbar/xdg-data-dirs.patch patch gonna stay ?

**Comment by JakeStanger** [comment]
I believe that patch is from a PR that's currently open, so should hopefully have that merged soon.

**Comment by donovanglover** [comment]
Ran `nixpkgs-fmt` since I forgot to check formatting earlier. Patch will be removed once https://github.com/JakeStanger/ironbar/pull/210 gets merged.

**Comment by yavko** [comment]
Hi as the author and essentially maintainer of the package, I just wanna say that there are some possibly unnecessary dependencies included with the GTK wrapper. Though I haven't been able to test for which ones aren't needed, for nixpkgs it should probably be cleaned up. Also I am willing to port the home manager module to NixOS if needed, as I also wrote it and know it quite well.

**Comment by donovanglover** [comment]
@yavko Thank you for all your work! This PR would not exist without your contributions. Let me know if you want to be added as a maintainer for the nixpkgs version as well.

I'll see if I can clean up the wrapper later. In the meantime, I added the upstream version to follow https://github.com/NixOS/nixpkgs/pull/234201

**Comment by yavko** [comment]
> @yavko Thank you for all your work! This PR would not exist without your contributions. Let me know if you want to be added as a maintainer for the nixpkgs version as well.

Sure, though I'm not a maintainer in nixpkgs

> I'll see if I can clean up the wrapper later. In the meantime, I added the upstream version to follow https://github.com/NixOS/nixpkgs/pull/234201

This is new to me, I'll have to start using this pattern lol

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/ironbar/default.nix
Comment (JakeStanger) [reply pkgs/applications/misc/ironbar/default.nix:74]: These were added as part of a fix for blurry images.  For flake/home manager at least, setting them did have an impact.  https://github.com/JakeStanger/ironbar/pull/122

Unsure as to what the expected behaviour here is.

Comment (yavko) [reply pkgs/applications/misc/ironbar/default.nix:74]: > These were added as part of a fix for blurry images.  For flake/home manager at least, setting them did have an impact.  https://github.com/JakeStanger/ironbar/pull/122
> 
> Unsure as to what the expected behaviour here is. 

The fix was to add the GTK wrapper, but the example for the wrapper on some docs (I think nixpkgs) showed it something like this.

File: pkgs/applications/misc/ironbar/default.nix
Comment (donovanglover) [reply pkgs/applications/misc/ironbar/default.nix:29]: I feel like it's better to wait for the next release of ironbar so more bugs can be fixed first. I've made this PR a draft for now.

File: pkgs/applications/misc/ironbar/default.nix
Comment (donovanglover) [reply pkgs/applications/misc/ironbar/default.nix:74]: I'll test without this part later to see if I notice a difference.

**Comment by donovanglover** [comment]
Added gtk3 to postInstall since ironbar [depends on](https://github.com/JakeStanger/ironbar/blob/fc820746a4eb583bcea59d1fc41adad7e43628b5/src/modules/launcher/mod.rs#L253) `gtk-launch`.

@yavko If you fork nixpkgs and make a commit adding yourself to https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix I should be able to add that commit to this PR. Format should be `maintainers: add name`.

**Comment by yavko** [comment]
> Added gtk3 to postInstall since ironbar [depends on](https://github.com/JakeStanger/ironbar/blob/fc820746a4eb583bcea59d1fc41adad7e43628b5/src/modules/launcher/mod.rs#L253) `gtk-launch`.
> 
> @yavko If you fork nixpkgs and make a commit adding yourself to https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix I should be able to add that commit to this PR. Format should be `maintainers: add name`.

Sorry for taking so long, but I was on a trip at the time and forgot https://github.com/NixOS/nixpkgs/pull/248376

**Comment by donovanglover** [comment]
LGTM, I'll rebase and update this later. Will also add you as a co-author since you wrote the original `flake.nix`.

**Review by yavko — APPROVED** [review]

File: pkgs/applications/misc/ironbar/default.nix
Comment (donovanglover) [reply pkgs/applications/misc/ironbar/default.nix:67]: @yavko So I don't notice a difference but I don't use hidpi. Can you confirm that this is still needed? If so, I'll need to figure something out to avoid the double wrapping with `wrapProgram` in `postInstall`.

**Review by yavko — APPROVED** [review]

File: pkgs/applications/misc/ironbar/default.nix
Comment (donovanglover) [reply pkgs/applications/misc/ironbar/default.nix:67]: Double wrapper should be fixed now (will assume that this is needed)

File: pkgs/applications/misc/ironbar/default.nix
Comment (JakeStanger) [inline pkgs/applications/misc/ironbar/default.nix:77]: I believe I am in the maintainers list somewhere as `jakestanger` if you want to add me as a 3rd member. I won't be able to be a "full" maintainer like you guys, but I'll be happy to have my name in the list and help out wherever I can.

Comment (donovanglover) [reply pkgs/applications/misc/ironbar/default.nix]: Done

**Review by yavko — APPROVED** [review]

**Review by JakeStanger — APPROVED** [review]

**Comment by Janik-Haag** [comment]
Would you mind switching to the new [pkgs/by-name](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name) scheme?

**Review by kip93 — APPROVED** [review]

**Comment by donovanglover** [comment]
Looks like everything should be good to go now. I tested ironbar on master. Excited to see it in nixpkgs!

**Pull request merged**
